### ISSUE TITLE
Undo Tests Fail for Can_Undo Issue

### ISSUE THREAD

**Issue by danyeaw** [gaphor/gaphor#75]
Two tests are failing in the UndoRedoTestCase and the UndoTest for a similar issue for a failed assert:
`self.assertTrue(undo_manager.can_undo())`

### I'm submitting a...
- [X] Bug report  
- [ ] Feature request
- [ ] Documentation issue or request

### Current behavior

```
tests/test_undo.py:11 (UndoTest.test_class_association_undo_redo)
self = <test_undo.UndoTest testMethod=test_class_association_undo_redo>
def test_class_association_undo_redo(self):
factory = self.element_factory
undo_manager = self.get_service("undo_manager")

self.assertEqual(0, len(self.diagram.canvas.solver.constraints))

ci1 = self.create(items.ClassItem, UML.Class)
self.assertEqual(2, len(self.diagram.canvas.solver.constraints))

ci2 = self.create(items.ClassItem, UML.Class)
self.assertEqual(4, len(self.diagram.canvas.solver.constraints))

a = self.create(items.AssociationItem)

self.connect(a, a.head, ci1)
self.connect(a, a.tail, ci2)

# Diagram, Association, 2x Class, Property, LiteralSpecification
self.assertEqual(6, len(factory.lselect()))
self.assertEqual(6, len(self.diagram.canvas.solver.constraints))

@transactional
def delete_class():
ci2.unlink()

undo_manager.clear_undo_stack()
self.assertFalse(undo_manager.can_undo())

delete_class()

>       self.assertTrue(undo_manager.can_undo())
E AssertionError: False is not true
tests/test_undo.py:42: AssertionError
```

**Comment by amolenaar** [comment]
Annoyingly, this test passes when run in isolation.

**Comment by amolenaar** [comment]
I can only trigger the issue when combining `gaphor/tests` and `tests/test_undo.py` in one test run.

**Comment by amolenaar** [comment]
Fixed in #91
